metrics/internal/factory/factory.go

```diff
@@ -0,0 +1,68 @@
+package factory
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/mozilla-services/rapid-release-model/metrics/internal/config"
+	"github.com/mozilla-services/rapid-release-model/metrics/internal/export"
+	"github.com/mozilla-services/rapid-release-model/metrics/internal/github"
+	"github.com/shurcooL/githubv4"
+	"golang.org/x/oauth2"
+)
+
+// Factory is used to create dependencies for the CLI application
+type Factory struct {
+	NewEncoder             func() (export.Encoder, error)
+	NewExporter            func() (export.Exporter, error)
+	NewGitHubHTTPClient    func() (*http.Client, error)
+	NewGitHubGraphQLClient func() (github.GraphQLClient, error)
+	NewGitHubRepo          func() (*github.Repo, error)
+}
+
+// NewFactory creates the default Factory for the CLI application
+func NewFactory(ctx context.Context) *Factory {
+	f := new(Factory)
+
+	f.NewEncoder = func() (export.Encoder, error) {
+		return &export.JSONEcoder{}, nil
+	}
+
+	f.NewExporter = func() (export.Exporter, error) {
+		encoder, err := f.NewEncoder()
+		if err != nil {
+			return nil, err
+		}
+		return export.NewWriterExporter(os.Stdout, encoder)
+	}
+
+	f.NewGitHubHTTPClient = func() (*http.Client, error) {
```

Comment:
We may need to add GitHub app auth if we want to run this regularly against our repos. 

https://github.com/int128/oauth2-github-app

https://docs.github.com/en/apps/creating-github-apps/writing-code-for-a-github-app/building-a-cli-with-a-github-app#introduction

Reply:
That sounds reasonable! 👍🏻